src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,851 @@
+"""Class to download and process USGS FEWS NET NDVI data.
+
+Data is downloaded from the `USGS FEWS NET data portal
+<https://earlywarning.usgs.gov/fews>`_. Data is
+generated from eMODIS AQUA, with full methodological
+details available on the `Documentation page
+<https://earlywarning.usgs.gov/fews/product/449#documentation>`_
+for the specific product. The available areas of
+coverage are:
+
+- `North Africa<https://earlywarning.usgs.gov/fews/product/449>`_
+- `East Africa<https://earlywarning.usgs.gov/fews/product/448>`_
+- `Southern Africa<https://earlywarning.usgs.gov/fews/product/450>`_
+- `West Africa<https://earlywarning.usgs.gov/fews/product/451>`_
+- `Central Asia<https://earlywarning.usgs.gov/fews/product/493>`_
+- `Yemen<https://earlywarning.usgs.gov/fews/product/502>`_
+- `Central America<https://earlywarning.usgs.gov/fews/product/445>`_
+- `Hispaniola<https://earlywarning.usgs.gov/fews/product/446>`_
+
+Data is made available on the backend USGS file explorer. For example,
+dekadal temporally smooth NDVI data for West Africa is available at
+`this link
+<https://edcintl.cr.usgs.gov/downloads/sciweb1/shared/fews/web/africa/west/dekadal/emodis/ndvi_c6/temporallysmoothedndvi/downloads/monthly/>`_
+
+The products include temporally smoothed NDVI, median anomaly,
+difference from the previous year, and median anomaly
+presented as a percentile.
+
+Data by USGS is published quickly after the dekad.
+After about 1 month this data is updated with temporal smoothing
+and error correction for cloud cover. Files for a specific
+dekad and region can range from 30MB up to over 100MB, so
+downloading and processing can take a long time.
+"""
+
+# TODO: add progress bar
+import functools
+import logging
+import re
+from datetime import date, datetime
+from io import BytesIO
+from pathlib import Path
+from typing import Generator, List, Optional, Tuple, Union
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import geopandas as gpd
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+from rasterio.errors import RasterioIOError
+
+import aatoolbox.utils.raster  # noqa: F401
+from aatoolbox.config.countryconfig import CountryConfig
+from aatoolbox.datasources.datasource import DataSource
+from aatoolbox.utils.dates import (
+    _compare_dekads_gt,
+    _compare_dekads_lt,
+    _dekad_to_date,
+    _expand_dekads,
+    _get_dekadal_date,
```

Comment:
Sorry, I can't find the conversation I saw you had with Tinka on this, but if I recall correctly there was a link to an SO post, and one of the answers basically said that if you want modules to not be part of the API, just don't put them in `__init__`.  

The leading underscore for a function denotes that it's internal within its own module. So if you're going to import these functions in other modules, the leading underscores should be removed.

Reply:
Yeah, agreed. In that case, should we make the module "private" then with `_dates.py`? That was based on [this post](https://stackoverflow.com/questions/3602110/python-private-module-in-a-package).

Reply:
No, neither the module or the functions should be private, as they are used in other modules. But they are "hidden" from the user by not being in the top level __init__.